Tasktive/Models/ViewModels/TasksViewModel.swift

```diff
@@ -176,6 +176,72 @@ final class TasksViewModel: ObservableObject {
         })
     }
 
+    func deleteTasks(by date: Date, indexSet: IndexSet) async -> Result<Void, UserErrors> {
+        let dateHash = getHashDate(from: date)
+        var mutableTask = tasks
+        let indicesReversed = indexSet.sorted().reversed()
+
+        for taskIndex in indicesReversed {
+            let id = tasks[dateHash]![taskIndex].id
+            _ = await deleteTask(id, onSuccess: { })
+        }
+
+        for taskIndex in indicesReversed {
+            mutableTask[dateHash]?.remove(at: taskIndex)
+        }
+
+        await setTasks(mutableTask[dateHash] ?? [], forDate: dateHash)
+
+        return .success(())
+    }
+
+    func deleteTask(on date: Date, by id: UUID) async -> Result<Void, UserErrors> {
+        await withSettingTasks(completion: {
+            let dateHash = getHashDate(from: date)
+            guard let taskIndex = tasks[dateHash]?.findIndex(by: \.id, is: id)
+            else { return .failure(.deleteFailure) }
+
+            return await deleteTask(id, onSuccess: {
+                var mutableTask = tasks
+                mutableTask[dateHash]?.remove(at: taskIndex)
+                await setTasks(mutableTask[dateHash] ?? [], forDate: dateHash)
+            })
+        })
+    }
+
+    private func deleteTask(_ id: UUID, onSuccess: () async -> Void) async -> Result<Void, UserErrors> {
```

Comment:
I don't see the need to use `onSuccess` here, I don't see it being handled anywhere necessarily. If `await` for this method than it already indicates that the method has finished.